Src/GoogleApis.Auth/OAuth2/ClientCredentialParameters.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+Copyright 2015 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Google.Apis.Json;
+using Google.Apis.Util;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>Indicates the type of Client Credential.</summary>
+    public enum ClientCredentialType
+    {
+        /// <summary>UserCredential is created by the gcloud sdk tool when the user runs <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        UserCredential = 1,
+        /// <summary>ServiceAccountCredential is downloaded by the user from <a href="https://console.developers.google.com">Google Developers Console</a>.</summary>
+        ServiceAccountCredential = 2
+    }
+
+    /// <summary>Container for the union of parameters/properties from all credential types.</summary>
+    public sealed class ClientCredentialParameters
+    {
+        /// <summary>Indicates whether the credentials is of type ServiceAccountCredential or UserCredential.</summary>
+        [Newtonsoft.Json.JsonProperty("type")]
+        private string Type { get; set; }
```

Comment:
The ClientCredentialParameters is a data holder object (an entity that has more or less 1:1 mapping to JSON repr). For such an object, the concept of making a property private doesn't make sense. Someone working with such object should be able to access its properties (how else would he know which fields to access if there's no way to tell the type of this object).